activerecord/lib/active_record/connection_adapters/abstract/connection_pool.rb

```diff
@@ -1097,15 +1103,15 @@ def flush_idle_connections!
       # active or defined connection: if it is the latter, it will be
       # opened and set as the active connection for the class it was defined
       # for (not necessarily the current class).
-      def retrieve_connection(spec_name) #:nodoc:
-        pool = retrieve_connection_pool(spec_name)
+      def retrieve_connection(role) #:nodoc:
+        pool = retrieve_connection_pool(role)
 
         unless pool
           # multiple database application
-          if ActiveRecord::Base.connection_handler != ActiveRecord::Base.default_connection_handler
-            raise ConnectionNotEstablished, "No connection pool with '#{spec_name}' found for the '#{ActiveRecord::Base.current_role}' role."
+          if connection = ActiveRecord::Base.connection_handler.connection_pool_list.first
```

Comment:
Thanks for pointing that out, just realised it it wasn't 100% correct.

> If there's a pool in the list and we have a connection then why does this end up raising?

It end up raising because there is no pool for the given role. What I want to do here is grab any connection pool (doesn't matter which role) in order to get the database name. This comes with the assumptions that each roles (writing, reading) on the handler will connect to the same db